### DISCUSSION TITLE
Photo Metadata for random image

### DISCUSSION THREAD

**Discussion by imaginator** [bpatrik/pigallery2#595] [Q&A] [answered]
I use pigallery2 primarily to serve up a random image as the [screensaver to my wall mounted home automation tablets](https://github.com/imaginator/smartserver/blob/master/roles/homeassistant/templates/index.html). This works fantastically well.

I'd like to also show the photo's capture/digitization/whatever-date (or any other photo metadata) Is there a method to also retrieve the metadata for the random image or to have this returned in the header/somehow?

**Comment by bpatrik** [comment, marked as answer]
Hi

The app only returns with the original photo.
Given that you can use any 3rd party application to parse that image and
load it's metadata.

-- Sorry for being brief, sent from my phone.

**Reply by imaginator** [reply]
I'd just assumed that it wasn't possible to read EXIF data in the browser but as you point out, it seems to be. Thanks!

**Comment by bpatrik** [comment]
There are a lot of JavaScript libraries that can read exif and iptc data.

Here is one that I'm considering using for pigallery:
https://mutiny.cz/exifr/

-- Sorry for being brief, sent from my phone.

**Reply by imaginator** [reply]
That's a super nice library. Thanks.

It was all going to be easy but then I discovered that [nginx's image resizing](https://github.com/imaginator/smartserver/blob/master/roles/homeassistant/templates/ha.imagilan.conf#L13-L16) strips the metadata. Such is software :)

**Comment by bpatrik** [comment]
Im not familiar with that repo, but this file can be updated:
https://github.com/imaginator/smartserver/blob/master/roles/pigallery2/templates/pigallery2.imagilan.conf

The pigallery2 recommend nginx config only allows post/put to pgapi/ path
and it also enables gzip compression (that gives a significant speed up)

-- Sorry for being brief, sent from my phone.

**Reply by imaginator** [reply]
Thanks for the update - yes that nginx config was just a hacky way to resize photos before so that the underpowered wall-mounted tablet doesn't choke on large files :) It turns out nginx has a nice way to do this. Or is there a param to pass to `/api/gallery/random` to resize the image?
